File: java_sdk/src/main/java/business_api_client/MemberInviteBody.java
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */

package business_api_client;

import java.util.Objects;
import java.util.Arrays;
import business_api_client.BcmemberinviteExtUserRole;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.v3.oas.annotations.media.Schema;
import java.util.ArrayList;
import java.util.List;
/**
 * MemberInviteBody
 */

@javax.annotation.Generated(value = "com.tiktok.codegen.JavatiktokcodegenGenerator", date = "2025-05-20T12:00:30.004772-07:00[America/Los_Angeles]")
public class MemberInviteBody {
  @JsonProperty("advertiser_role")
  private String advertiserRole = "ANALYST";

  @JsonProperty("asset_ids")
  private List<String> assetIds = new ArrayList<String>();

  @JsonProperty("asset_type")
  private String assetType = "ADVERTISER";

  @JsonProperty("bc_id")
  private String bcId = null;

  @JsonProperty("emails")
  private List<String> emails = new ArrayList<String>();

  @JsonProperty("ext_user_role")
  private BcmemberinviteExtUserRole extUserRole = null;

  @JsonProperty("user_role")
  private String userRole = "STANDARD";

  public MemberInviteBody advertiserRole(String advertiserRole) {
    this.advertiserRole = advertiserRole;
    return this;
  }

   /**
   * Get advertiserRole
   * @return advertiserRole
  **/
  @Schema(description = "")
  public String getAdvertiserRole() {
    return advertiserRole;
  }

  public void setAdvertiserRole(String advertiserRole) {
    this.advertiserRole = advertiserRole;
  }

  public MemberInviteBody assetIds(List<String> assetIds) {
    this.assetIds = assetIds;
    return this;
  }

  public MemberInviteBody addAssetIdsItem(String assetIdsItem) {
    this.assetIds.add(assetIdsItem);
    return this;
  }

   /**
   * Get assetIds
   * @return assetIds
  **/
  @Schema(required = true, description = "")
  public List<String> getAssetIds() {
    return assetIds;
  }

  public void setAssetIds(List<String> assetIds) {
    this.assetIds = assetIds;
  }

  public MemberInviteBody assetType(String assetType) {
    this.assetType = assetType;
    return this;
  }

   /**
   * Get assetType
   * @return assetType
  **/
  @Schema(description = "")
  public String getAssetType() {
    return assetType;
  }

  public void setAssetType(String assetType) {
    this.assetType = assetType;
  }

  public MemberInviteBody bcId(String bcId) {
    this.bcId = bcId;
    return this;
  }

   /**
   * Get bcId
   * @return bcId
  **/
  @Schema(required = true, description = "")
  public String getBcId() {
    return bcId;
  }

  public void setBcId(String bcId) {
    this.bcId = bcId;
  }

  public MemberInviteBody emails(List<String> emails) {
    this.emails = emails;
    return this;
  }

  public MemberInviteBody addEmailsItem(String emailsItem) {
    this.emails.add(emailsItem);
    return this;
  }

   /**
   * Get emails
   * @return emails
  **/
  @Schema(required = true, description = "")
  public List<String> getEmails() {
    return emails;
  }

  public void setEmails(List<String> emails) {
    this.emails = emails;
  }

  public MemberInviteBody extUserRole(BcmemberinviteExtUserRole extUserRole) {
    this.extUserRole = extUserRole;
    return this;
  }

   /**
   * Get extUserRole
   * @return extUserRole
  **/
  @Schema(description = "")
  public BcmemberinviteExtUserRole getExtUserRole() {
    return extUserRole;
  }

  public void setExtUserRole(BcmemberinviteExtUserRole extUserRole) {
    this.extUserRole = extUserRole;
  }

  public MemberInviteBody userRole(String userRole) {
    this.userRole = userRole;
    return this;
  }

   /**
   * Get userRole
   * @return userRole
  **/
  @Schema(description = "")
  public String getUserRole() {
    return userRole;
  }

  public void setUserRole(String userRole) {
    this.userRole = userRole;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    MemberInviteBody memberInviteBody = (MemberInviteBody) o;
    return Objects.equals(this.advertiserRole, memberInviteBody.advertiserRole) &&
        Objects.equals(this.assetIds, memberInviteBody.assetIds) &&
        Objects.equals(this.assetType, memberInviteBody.assetType) &&
        Objects.equals(this.bcId, memberInviteBody.bcId) &&
        Objects.equals(this.emails, memberInviteBody.emails) &&
        Objects.equals(this.extUserRole, memberInviteBody.extUserRole) &&
        Objects.equals(this.userRole, memberInviteBody.userRole);
  }

  @Override
  public int hashCode() {
    return Objects.hash(advertiserRole, assetIds, assetType, bcId, emails, extUserRole, userRole);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class MemberInviteBody {\n");
    
    sb.append("    advertiserRole: ").append(toIndentedString(advertiserRole)).append("\n");
    sb.append("    assetIds: ").append(toIndentedString(assetIds)).append("\n");
    sb.append("    assetType: ").append(toIndentedString(assetType)).append("\n");
    sb.append("    bcId: ").append(toIndentedString(bcId)).append("\n");
    sb.append("    emails: ").append(toIndentedString(emails)).append("\n");
    sb.append("    extUserRole: ").append(toIndentedString(extUserRole)).append("\n");
    sb.append("    userRole: ").append(toIndentedString(userRole)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}

File: js_sdk/src/model/MemberInviteBody.js
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */
import {ApiClient} from '../ApiClient.js';
import {BcmemberinviteExtUserRole} from './BcmemberinviteExtUserRole.js';

/**
 * The MemberInviteBody model module.
 * @module model/MemberInviteBody
 * @version 0.1.8
 */
export class MemberInviteBody {
  /**
   * Constructs a new <code>MemberInviteBody</code>.
   * @alias module:model/MemberInviteBody
   * @class
   * @param asset_ids {Array.<String>} 
   * @param bc_id {String} 
   * @param emails {Array.<String>} 
   */
  constructor(asset_ids, bc_id, emails) {
    this.asset_ids = asset_ids;
    this.bc_id = bc_id;
    this.emails = emails;
  }

  /**
   * Constructs a <code>MemberInviteBody</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/MemberInviteBody} obj Optional instance to populate.
   * @return {module:model/MemberInviteBody} The populated <code>MemberInviteBody</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new MemberInviteBody();
      if (data.hasOwnProperty('advertiser_role'))
        obj.advertiser_role = ApiClient.convertToType(data['advertiser_role'], 'String');
      if (data.hasOwnProperty('asset_ids'))
        obj.asset_ids = ApiClient.convertToType(data['asset_ids'], ['String']);
      if (data.hasOwnProperty('asset_type'))
        obj.asset_type = ApiClient.convertToType(data['asset_type'], 'String');
      if (data.hasOwnProperty('bc_id'))
        obj.bc_id = ApiClient.convertToType(data['bc_id'], 'String');
      if (data.hasOwnProperty('emails'))
        obj.emails = ApiClient.convertToType(data['emails'], ['String']);
      if (data.hasOwnProperty('ext_user_role'))
        obj.ext_user_role = BcmemberinviteExtUserRole.constructFromObject(data['ext_user_role']);
      if (data.hasOwnProperty('user_role'))
        obj.user_role = ApiClient.convertToType(data['user_role'], 'String');
    }
    return obj;
  }
}

/**
 * @member {String} advertiser_role
 * @default 'ANALYST'
 */
MemberInviteBody.prototype.advertiser_role = 'ANALYST';

/**
 * @member {Array.<String>} asset_ids
 */
MemberInviteBody.prototype.asset_ids = undefined;

/**
 * @member {String} asset_type
 * @default 'ADVERTISER'
 */
MemberInviteBody.prototype.asset_type = 'ADVERTISER';

/**
 * @member {String} bc_id
 */
MemberInviteBody.prototype.bc_id = undefined;

/**
 * @member {Array.<String>} emails
 */
MemberInviteBody.prototype.emails = undefined;

/**
 * @member {module:model/BcmemberinviteExtUserRole} ext_user_role
 */
MemberInviteBody.prototype.ext_user_role = undefined;

/**
 * @member {String} user_role
 * @default 'STANDARD'
 */
MemberInviteBody.prototype.user_role = 'STANDARD';

